Task 3/Task 3.3/Task 3.3.2/SuperString/SuperString.cs

```
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace SuperString
+{
+    public static class SuperString
+    {
+        public static TextType GetTextType(this string text)
+        {
+            if (text.All(c => char.IsDigit(c)))
+                return TextType.Number;
+
+            if (text.All(c => char.IsLetter(c)))
+            {
+                if (Regex.IsMatch(text, @"^[а-яА-ЯЁё]*$"))
+                    return TextType.Russian;
+                if (Regex.IsMatch(text, @"^[a-zA-Z]*$"))
+                    return TextType.English;
+            }
+
+            return TextType.Mixed;
+        }
```

Regex is cool and all, but in the worst case you iterate through text 4 times. You can do it in one enumeration (and fail fast)